src/Clay/Grid.hs

```diff
@@ -1,58 +1,249 @@
+rowGap :: Size a -> Css
+rowGap = key "row-gap" <> key "grid-row-gap"
+
+-- | Property sets the size of the gap (gutter) between an element's grid columns.
+-- Sets both "column-gap" & "grid-column-gap"
+-- to quote: https://developer.mozilla.org/en-US/docs/Web/CSS/column-gap
+-- "CSS Grid Layout initially defined the grid-column-gap property. This prefixed property is being replaced by column-gap. However, in order to support bcolumnsers that implemented grid-column-gap and not column-gap for grid, you will need to use the prefixed property."
+columnGap :: Size a -> Css
+columnGap = key "column-gap" <> key "grid-column-gap"
+
+-------------------------------------------------------------------------------
+
+-- | Property defines the line names and track sizing functions of the grid rows.
+gridTemplateRows :: GridTrackList a -> Css
+gridTemplateRows = key "grid-template-rows"
+
+-- | Property defines the line names and track sizing functions of the grid columns.
+gridTemplateColumns :: GridTrackList a -> Css
+gridTemplateColumns = key "grid-template-columns"
+
+newtype GridTrackList a = GridTrackList Value
+  deriving (Val, None, Inherit, Initial, Unset)
+
+-- | Create a GridTrackList from a list of sizes
+mkGridTrackList :: [Size a] -> GridTrackList a
+mkGridTrackList = GridTrackList . noCommas
+
+-------------------------------------------------------------------------------
+
+-- | Property defines the line names and track sizing functions of the grid rows.
+gridAutoRows :: GridAutoTrackList a -> Css
+gridAutoRows = key "grid-auto-rows"
 
 -- | Property defines the line names and track sizing functions of the grid columns.
-gridTemplateColumns :: [Size a] -> Css
-gridTemplateColumns = key "grid-template-columns" . noCommas
+gridAutoColumns :: GridAutoTrackList a -> Css
+gridAutoColumns = key "grid-auto-columns"
+
+newtype GridAutoTrackList a = GridAutoTrackList Value
+  deriving (Val, Auto, MinContent, MaxContent, Inherit, Initial, Unset)
+
+mkGridAutoTrackList :: [Size a] -> GridAutoTrackList
+mkGridAutoTrackList = GridAutoTrackList . noCommas
+
+-------------------------------------------------------------------------------
+gridAutoFlow :: GridAutoFlow -> Css
+gridAutoFlow = key "grid-auto-flow"
+
+newtype GridAutoFlow = GridAutoFlow Value
+  deriving (Val, Row, Column, Inherit, Initial, Unset)
+
+dense :: GridAutoFlow
+dense = GridAutoFlow "dense"
+
+rowDense :: GridAutoFlow
+rowDense = GridAutoFlow "row dense"
+
+columnDense :: GridAutoFlow
+columnDense = GridAutoFlow "column dense"
+-------------------------------------------------------------------------------
+-- | Property defines the element location inside grid template
+gridArea :: GridArea -> Css
+gridArea = key "grid-area"
+
+blankGridArea :: GridArea
+blankGridArea = GridArea "."
+
+newtype GridArea = GridArea Text
+  deriving (IsString, Val)
+
+-------------------------------------------------------------------------------
+
+gridRowStart :: GridLocation -> Css
```

Comment:
But I thought grid rows don't support spans? See https://www.w3schools.com/cssref/pr_grid-row-start.asp. Maybe `GridLocation` should be `GridColumnLocation`?